apps/web/pages/getting-started/[[...step]].tsx

```diff
@@ -104,6 +158,242 @@ const OnboardingPage = () => {
 
   const currentStepIndex = steps.indexOf(currentStep);
 
+  function isPointInsidePolygon(point: number[], polygon: ([number, number] | null | undefined)[]) {
+    const x = point[0];
+    const y = point[1];
+
+    let inside = false;
+    for (let i = 0, j = polygon.length - 1; i < polygon.length; j = i++) {
+      const xi = polygon[i]?.[0];
+      const yi = polygon[i]?.[1];
+      const xj = polygon[j]?.[0];
+      const yj = polygon[j]?.[1];
+
+      if (xi == undefined || yi == undefined || xj == undefined || yj == undefined) continue;
+
+      const intersect = yi > y !== yj > y && x < ((xj - xi) * (y - yi)) / (yj - yi) + xi;
+
+      if (intersect) inside = !inside;
+    }
+
+    return inside;
+  }
+
+  useEffect(() => {
+    if (timeZone !== "" && timeZone !== null) {
+      // Search country by timezone
+      const targetCountry: ICountry | undefined = Country.getAllCountries().find((c: any) => {
+        const alltimezones = c.timezones.map((t: any) => {
+          return t.zoneName;
+        });
+
+        if (alltimezones.includes(timeZone)) return true;
+
+        return false;
+      });
+
+      if (!targetCountry) return;
+
+      // Find country's geo json data
+      const country = worldviewGeoJSON.features.find((feature) => {
+        return feature.properties.name_en === targetCountry.name;
+      });
+      const areaCoordinates: ([number, number] | null)[] = [];
+      country?.geometry.coordinates.map((g) => {
+        areaCoordinates.push(...(g as [number, number][]));
+      });
+
+      // Convert geo json data to svg coordinates
+      let svgCoordinates: ([number, number] | null | undefined)[] = [];
+      let countryCoord: [number, number] = [0, 0];
+      if (areaCoordinates) {
+        const projection = d3.geoMercator().scale(100).translate([318, 235]);
+
+        svgCoordinates = areaCoordinates
+          .filter((coord) => coord !== null)
+          .map((coord) => {
+            if (coord) {
+              return projection(coord);
+            }
+          });
+      }
```

Comment:
This could easily be a backend endpoint so we only return the data needed instead of download the whole "country-state-city" data in the JS bundle.